internal/annotations/annotations.go

```diff
@@ -381,3 +383,8 @@ func ExtractRewriteURI(anns map[string]string) (string, bool) {
 	s, ok := anns[AnnotationPrefix+RewriteURIKey]
 	return s, ok
 }
+
+func ExtractUpstreamPolicy(anns map[string]string) (string, bool) {
+	s, ok := anns[kongv1beta1.KongUpstreamPolicyAnnotationKey]
```

Comment:
Why define the annotation key in `kongv1beta1` package? The annotation keys are usually defined in this file.

Reply:
I believe we should do so because it's part of our public interface. We could move our other annotations as well to `apis`.

Reply:
Yeah, this makes sense, these should be public, we just have long-standing debt of making most code internal for no particular reason.

I'm just gonna just go ahead and move it, though to a dedicated annotations package still, mostly because we haven't been versioning these like CRD APIs. We may consider doing that also, but for now simply moving them to a public package is simpler: https://github.com/Kong/kubernetes-ingress-controller/issues/5020

Reply:
> I'm just gonna just go ahead and move it, though to a dedicated annotations package still, mostly because we haven't been versioning these like CRD APIs

I'd consider putting those annotations in the versioned package(s) though, next to resources that would use those. This way it will be more discoverable (you'd have them next to resources that use them and in a shared file for those that can be used with multiple types) and easier to version those if we ever desire to (no need to do `Name` and then `NameLegacy` but proper `v1`, `v2alpha2` etc)

Reply:
As we've created a separate issue (https://github.com/Kong/kubernetes-ingress-controller/issues/5020) to move annotations to public packages, and putting the annotation in this PR already under `apis/` is on par with that, I conclude we may resolve this conversation.